app/src/main/java/ai/brokk/context/DiffService.java

```diff
@@ -136,29 +137,23 @@ private static boolean isNewFileInGit(ContextFragment fragment, IGitRepo repo) {
      */
     @Blocking
     public static List<Context.DiffEntry> computeDiff(Context curr, Context other) {
-        try {
-            // Candidates:
-            // - Project text files that are editable (preserves read-only exclusion).
-            // - Git file fragments.
-            // - Image fragments (non-text), including pasted images and image files.
-            var projectPathEditable =
-                    curr.getEditableFragments().filter(f -> f.getType() == ContextFragment.FragmentType.PROJECT_PATH);
-            var gitFileFragments =
-                    curr.allFragments().filter(f -> f.getType() == ContextFragment.FragmentType.GIT_FILE);
-            var imageFragments = curr.allFragments().filter(f -> !f.isText());
-
-            var diffFutures = Stream.concat(Stream.concat(projectPathEditable, gitFileFragments), imageFragments)
-                    .map(cf -> computeDiffForFragment(curr, cf, other))
-                    .toList();
-
-            return diffFutures.stream()
-                    .map(CompletableFuture::join)
-                    .filter(Objects::nonNull)
-                    .toList();
-        } catch (Exception ex) {
-            logger.error("Error computing diffs between contexts: {}", ex.getMessage(), ex);
-            return List.of();
-        }
+        // Candidates:
```

Comment:
I think this was quite weird, so just included all editable fragments. Next path that we should ask is if Git file fragments should be here - sorta makes sense?

Reply:
Previous iterations of getting GPT on this got confused with "editable" fragments and "readonly" fragments so there were some weird candidate selection logics applied.